src/test/scala/PSO/ParticleSpec.scala

```diff
@@ -46,17 +46,88 @@ class ParticleSpec
     objectiveFunction = obj,
     initialSolutions = solutions,
     initialVelocity = Some(List(5.0, 5.0)),
-    topology = ParticleTopology.FullyConnected
+    topology = ParticleTopology.FullyConnected,
+    terminationCriteria = TerminationCriteria(maxIterations = 1)
   )
   val psoSystemFactory = new PSOSystemFactory(conf)
   val pso = psoSystemFactory.build(randomSeed = Some(0))
   val particles = pso.particles
 
-  override def afterAll {
+
+  val masterRef: TestActorRef[SwarmMaster] = TestActorRef(
+    Props(new SwarmMaster(
+      X => SampleFunctions.sphere(X.asInstanceOf[List[Double]]),
+      ProblemType.Minimization,
+      TerminationCriteria()
+    ))
+  )
+  val master = masterRef.underlyingActor
+  val particleRef: TestActorRef[Particle] = TestActorRef(Props(new Particle(
+    List(1.0, 1.0),
+    0.792,
+    1.4944,
+    1.4944,
+    List(-1.5, -0.5),
+    List(-0.1, 0.1),
+    ProblemType.Minimization,
+    SampleFunctions.sphere,
+    None,
+    masterRef,
+    0,
+    Some(0)
+  )))
+  val particle = particleRef.underlyingActor
+
+  override def afterAll() {
     shutdown()
   }
 
-  "A particle" should {
+  "A single particle" should {
+    "calculate a new velocity" in {
+      particle.calculateNewVelocity()
+      assert(particle.currVel == List(1.2952406674112558, 1.2855438362686753))
+    }
+
+    "calculate a new position" in {
+      particle.calculateNewPosition()
+      assert(particle.currPos == List(-0.20475933258874424, 0.7855438362686753))
+    }
+
+    "update local best if necessary" in {
```

Comment:
When is it necessary? I feel like you should define the scenario when it is necessary, and then just check that the appropriate function was invoked using `verify()`